staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/roundtrip_test.go

```diff
@@ -32,6 +32,11 @@ func TestRoundtrip(t *testing.T) {
 		dec cbor.DecMode
 	}
 
+	var absentSlice []interface{}
+	var absentMap map[string]interface{}
+	var nilPointerToSlice *[]interface{}
+	var nilPointerToMap *map[string]interface{}
```

Comment:
This highlights how awkward it is to write a Go expression for a nil pointer value with an arbitrary type (like the unreadable `[1]*[]interface{}{}[0]`). It would be ideal if people reading the test did not need to jump to these variables to understand the test cases that use them, though, and I expect that these won't be the last test cases to need to do this.

What do you think about adding a function above TestRoundtrip like this?

```go
func nilPointerFor[T interface{}]() *T {
	return nil
}
```

The two test cases could call `nilPointerFor[[]interface{}]()` and `nilPointerFor[map[string]interface{}]()` without needing separate variable declarations.

Reply:
Thanks for this wonderful suggestion! I've incorporated it.